File: packages/genai-rewriting/ios/Plugin/GenAiRewritingPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(GenAiRewritingPlugin, "GenAiRewriting",
           CAP_PLUGIN_METHOD(checkFeatureStatus, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(downloadFeature, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(rewrite, CAPPluginReturnPromise);
)

File: packages/genai-rewriting/ios/Plugin/GenAiRewritingPlugin.swift
import Foundation
import Capacitor

@objc(GenAiRewritingPlugin)
public class GenAiRewritingPlugin: CAPPlugin {
    @objc func checkFeatureStatus(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc func downloadFeature(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    @objc func rewrite(_ call: CAPPluginCall) {
        rejectCallAsUnavailable(call)
    }

    private func rejectCallAsUnavailable(_ call: CAPPluginCall) {
        call.unavailable("This method is not available on this platform.")
    }
}
